### ISSUE TITLE
hangs in training

### ISSUE THREAD

**Issue by fitsumreda** [roytseng-tw/Detectron.pytorch#30]
Thanks for your codes!
I was able to successfully train `configs/e2e_mask_rcnn_R-50-FPN_1x.yaml` with a (batch_size, learning_rate) = (8, 0.01) until a certain number of iterations (max = ~60K). So far, the losses look quite similar to your [benchmark](https://github.com/roytseng-tw/Detectron.pytorch#mask_rcnn). 
Training speed is also quite comparable to [Detectron](https://github.com/facebookresearch/Detectron)
The issue I'm having is the training hangs randomly at a certain iteration, which is not consistent from run to run, sometimes after 5K, 1K, or 60K iterations.
I'm using 4 V-100 GPUs.

Any thoughts?

**Comment by fitsumreda** [comment]
Thanks @roytseng-tw

I now used 4 P100 GPUs and the training completed successfully, with no hangs. However, the box and mask APs were slightly lower. I used a batch size of 8 and a learning rate of 0.02. Number of iterations is 90K, with learning rate down scaled at 60k and 80k iterations.

I used the training script in tools/train_net.py.

Any suggested optimal hyper parameters that would reproduce detection results?

Thanks!

**Comment by roytseng-tw** [comment]
Could you share your testing results ? and which architecture did you trained ?

**Comment by fitsumreda** [comment]
commit [9cbb285](https://github.com/roytseng-tw/Detectron.pytorch/commit/9cbb2855ba93239f5b085e4fbf283776decf9829)

I trained using `python tools/train_net_step.py --dataset coco2017 --cfg configs/e2e_mask_rcnn_R-50-FPN_1x.yaml --use_tfboard --bs 16 --nw 16` over 8 P100 GPUs, I got mAP 0.3598 for bbox, and  0.3252 for mask.

INFO task_evaluation.py:  65: Evaluating segmentations is done!
INFO task_evaluation.py: 180: copypaste: Dataset: coco_2017_val
INFO task_evaluation.py: 182: copypaste: Task: box
INFO task_evaluation.py: 185: copypaste: AP,AP50,AP75,APs,APm,APl
INFO task_evaluation.py: 186: copypaste: 0.3598,0.5697,0.3911,0.2030,0.3844,0.4752
INFO task_evaluation.py: 182: copypaste: Task: mask
INFO task_evaluation.py: 185: copypaste: AP,AP50,AP75,APs,APm,APl
INFO task_evaluation.py: 186: copypaste: 0.3252,0.5386,0.3419,0.1417,0.3454,0.4896

**Comment by JackHenry1992** [comment]
Yeah, I also get 0.357 mAP of bbox and 0.325 mAP of segm, which are slightly lower than 0.37 using 4 Tesla P40 GPUs with 16 batch size. The command is same as yours.

**Comment by roytseng-tw** [comment]
@fitsumreda, that's strange. Your results with 8 GPUs should be better than my results with only 4 GPUs reported in README, because I didn't adjust some RPN config options correctly before.

**Comment by roytseng-tw** [comment]
Close the issue, since it's off the topic.
Open a new issue if you still can't get the results right.
